pkg/server/server.go

```diff
@@ -66,12 +67,25 @@ func (s *CSIDriverProviderServer) Mount(ctx context.Context, req *v1alpha1.Mount
 	f := []*v1alpha1.File{}
 	// CSI driver v0.0.21+ will write to the filesystem if the files are in the response.
 	// No files in the response translates to "not implemented" in the CSI driver.
-	for k, v := range files {
-		f = append(f, &v1alpha1.File{
-			Path:     k,
-			Contents: v,
-			Mode:     int32(filePermission),
-		})
+	for _, file := range files {
+		secretFile := &v1alpha1.File{
+			Path:     file.Path,
+			Contents: file.Content,
+		}
+
+		if file.FileMode != "" {
```

Comment:
If we set the default file permission from driver when not overridden by user we can get rid of this check